src/cpp/r/RSexp.cpp

```diff
@@ -172,24 +196,58 @@ SEXP findVar(const std::string& name, const std::string& ns)
    
    return findVar(name, env);
 }
-  
+
 SEXP findFunction(const std::string& name, const std::string& ns) 
 {
+   r::sexp::Protect protect;
```

Comment:
Right -- my point is that even though the error doesn't cause a longjmp, it is still printed to the console, which is undesired for us here. E.g. some bad code:

```cpp
#include <Rcpp.h>
using namespace Rcpp;

void throw_error(void* data) {
    Rf_error("Ouch!");
}

// [[Rcpp::export]]
void test()
{
    R_ToplevelExec(throw_error, NULL);
    Rcout << "Still got here!" << std::endl;
}

/*** R
test()
*/
```

gives

```
> test()
Error: Ouch!
Still got here!
```